File: yandex_back_prep/leetcode/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t):
            return False
        
        cs = Counter(s)
        ct = Counter(t)

        for c in s:
            if c not in ct or ct[c] != cs[c]:
                return False
        
        for c in t:
            if c not in cs or cs[c] != ct[c]:
                return False

        return True

File: yandex_back_prep_python/leetcode_python/242. Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        map<char, int> counter1;
        map<char, int> counter2;

        for (auto c: s) {
            counter1[c] +=1;
        }

        for (auto c: t) {
            counter2[c] +=1;
        }

        return counter1 == counter2;